src/node.rs

```diff
@@ -776,7 +887,14 @@ impl<Lifetime, K, V> Handle<NodeRef<Lifetime, K, V, marker::Mut, marker::Interna
                 heap::deallocate(*right_node.node.ptr, mem::size_of::<LeafNode<K, V>>(), mem::align_of::<LeafNode<K, V>>());
             }
 
-            Handle::new(self.node, self.idx)
+            left_node.as_leaf_mut().len += right_len as u16 + 1;
+
+            if self.node.len() == 0 {
```

Comment:
I'd prefer for this logic to not be included in `merge` or anywhere else in the `node` module. My goal was for the `node` module to be completely oblivious to and not rely upon most of the important invariants of a BTree, including the idea that only the root can be underfull. Also, since memory safety relies on everything being constant depth, I'd like depth changes to be explicit.